File: android/DFM/src/main/java/com/dontflymoney/listeners/PickDate.java
package com.dontflymoney.listeners;

import android.app.DatePickerDialog;
import android.widget.DatePicker;

public class PickDate implements DatePickerDialog.OnDateSetListener
{
	private IDatePickerActivity activity;

	public PickDate(IDatePickerActivity activity)
	{
		this.activity = activity;
	}

	@Override
	public void onDateSet(DatePicker view, int year, int month, int day)
	{
		if (view.isShown())
		{
			activity.setResult(year, month, day);
			activity.getDialog().dismiss();
		}
	}
}

File: android/DFM/src/main/kotlin/com/darakeon/uiHelpers/listeners/PickDate.kt
package com.dontflymoney.listeners

import android.app.DatePickerDialog
import android.widget.DatePicker

class PickDate(private val activity: IDatePickerActivity) : DatePickerDialog.OnDateSetListener {

    override fun onDateSet(view: DatePicker, year: Int, month: Int, day: Int) {
        if (view.isShown) {
            activity.setResult(year, month, day)
            activity.dialog?.dismiss()
        }
    }
}
